src/mon/OSDMonitor.cc

```diff
@@ -7099,7 +7099,7 @@ static int parse_reweights(CephContext *cct,
       if (osd_weight.second.type() != json_spirit::str_type) {
 	return -EINVAL;
       }
-      auto weight = std::stoul(osd_weight.second.get_str());
+      auto weight = CEPH_OSD_IN*std::stod(osd_weight.second.get_str());
```

Comment:
should not multiply the weight with `CEPH_OSD_IN` again. and the weight is a fixed point number, not a float. please see https://github.com/ceph/ceph/blob/master/src/pybind/mgr/balancer/module.py#L961 .

Reply:
Thank you for the review and link. Now I see reweightn is intently used in balancer module.
Do you think moving the weight multiplier 0x10000 out of balancer/module.py could be a good idea? So end user would treat the weight value in osd reweight and osd reweightn as the same and be not confused.

Reply:
how do you plan to address the interoperability between new ceph-mgr and old ceph-mon?

Reply:
i guess that's a simpler alternative. but probably there could be some better approach(es) i don't know at this moment though..